content/general/events.textile

```diff
@@ -72,9 +72,276 @@ h3(#sources). Sources
 
 Ably currently supports three types of Events:
 
-* "Messages":#messages - messages trigger function calls as soon as they are published on a channel
+* "Messages":#messages - messages trigger function calls as soon as they are published on a channel. 
 * "Presence events":#presence - presence events trigger function calls when clients enter, update their data, or leave channels
-* "Channel lifecycle events":/realtime/channel-metadata – get notified when a channel is created (following the first client attaching to this channel) or discarded (when there are no more clients attached to the channel). Currently this is *only* available for "batched":#batching messages
+* "Channel lifecycle events":#lifecycle-events – get notified when a channel is created (following the first client attaching to this channel) or discarded (when there are no more clients attached to the channel). Currently this is *only* available for "batched":#batching messages
+
+h4(#messages). Messages
+
+If you are unfamiliar with how messages work in Ably, we recommend you first review the "realtime messages documentation":/realtime/messages.
+
+When a message is published on a channel, a message WebHook event is triggered.
+
+Every WebHook request will contain one or more enveloped messages in the root level attribute @items@, each of which in turn contains one or messages.
+
+Each message envelope in the @items@ array consists of an object with the following attributes:
+
+- name := the event type "@channel.message@"
+- webhookId := an internal unique ID for the configured WebHook
+- source := the source for the WebHook, namely "@channel.message@"
+- timestamp := a timestamp represented as milliseconds since epoch for the published message
+- data := an object containing the attributes defined below in "JSONPath format":http://goessner.net/articles/JsonPath/ @data.*@:
+- data.channelId := name of the channel that the presence event belongs to
+- data.site := an internal site identifier indicating the datacenter from which the message was published
+- data.messages := an @Array@ of messages with fields for each message described below
```

Comment:
Aren't these lists of fields now duplicated? Looks like Tom moved it down under "examples" and reformatted it a bit, and now you've re-added them at the top so they're there twice?

Reply:
Also it's very nonobvious that is the structure for batched payloads

Reply:
Ahh yes I can see this now looking at the deployment, I missed this as the examples wherein linked partial snippets that I couldn't see by just looking at the textile file itself. Do you think I should remove the parts that aren't about decoding specifically then? A little confused on the best resolution here.